mlir/include/mlir/Dialect/Bufferization/IR/BufferDeallocationOpInterface.h

```diff
@@ -106,6 +111,48 @@ struct DeallocationOptions {
   /// to, an error will already be emitted at compile time. This cannot be
   /// changed with this option.
   bool verifyFunctionBoundaryABI = true;
+
+  /// Given an allocation side-effect on the passed operation, determine whether
+  /// this allocation operation is of relevance (i.e., should assign ownership
+  /// to the allocated value). If it is determined to not be relevant,
+  /// ownership will be set to 'false', i.e., it will be leaked. This is useful
+  /// to support deallocation of multiple different kinds of allocation ops.
+  DetectionFn isRelevantAllocOp = [](Operation *op) {
+    return isa<memref::MemRefDialect, bufferization::BufferizationDialect>(
```

Comment:
Why is the bufferization dialect in here? Can we just check for memref.alloc/memref.alloca in here?

Reply:
`BufferizationDialect` is here because of `bufferization.clone` which has an allocation side-effect as well.